File: snake-strategy-bono/src/main/java/co/electric/snake/strategy/bonostrategy/distanceprocessor/WestLookingDistanceProcessor.java
package co.electric.snake.strategy.bonostrategy.distanceprocessor;

import co.electric.snake.framework.model.Coordinate;

public class WestLookingDistanceProcessor extends DistanceProcessor {

    public int getDistance(Coordinate actualCoordinate,
                           Coordinate blockingCoordinate, Coordinate maxCoordinate) {
        int distance = 0;

        if (actualCoordinate.getX() > blockingCoordinate.getX()) {
            distance = getDistanceDirectly(actualCoordinate.getX(),
                    blockingCoordinate.getX());
        }
        if (actualCoordinate.getX() < blockingCoordinate.getX()) {
            distance = getDistanceClipped(actualCoordinate.getX(),
                    blockingCoordinate.getX(), maxCoordinate.getX());
        }

        return distance;
    }

}

File: snake-strategy-bono/src/main/kotlin/co/electric/snake/strategy/bonostrategy/distanceprocessor/WestLookingDistanceProcessor.kt
package co.electric.snake.strategy.bonostrategy.distanceprocessor

import co.electric.snake.framework.model.Coordinate
import co.electric.snake.framework.model.Direction
import java.util.*

class WestLookingDistanceProcessor : DistanceProcessor {

    override fun getDistance(direction: Direction, headCoordinate: Coordinate, blockingCoordinate: Coordinate, maxCoordinate: Coordinate): Optional<Int> {
        var distance: Optional<Int> = Optional.empty()
        if (direction == Direction.WEST) {
            val calculatedDistance: Int
            if (headCoordinate.x >= blockingCoordinate.x) {
                calculatedDistance = getDistanceDirectly(headCoordinate.x, blockingCoordinate.x)
            } else {
                calculatedDistance = getDistanceClipped(headCoordinate.x, blockingCoordinate.x, maxCoordinate.x)
            }
            distance = Optional.of(calculatedDistance)
        }
        return distance
    }

}
